# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/midi/tests/browser_stable_midi_port_ids.js
# Commit: cfba299e5067
# Full Hash: cfba299e5067a4eab3f3b982374b0770970a3401
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-04-12 03:57:01
# Regressor Bug: 1758468
# File Overlap Count: 1
# Description:
#   Bug 1758468 - Use stable, anonymous IDs for Web MIDI ports r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142550
# ==============================================================================

diff -r c0eb987d0f76 -r cfba299e5067 dom/midi/tests/browser_stable_midi_port_ids.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/midi/tests/browser_stable_midi_port_ids.js	Mon Apr 11 17:43:09 2022 +0000
@@ -0,0 +1,52 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+const EXAMPLE_COM_URL = "https://example.com/browser/dom/midi/tests/";
+const EXAMPLE_ORG_URL = "https://example.org/browser/dom/midi/tests/";
+const PAGE1 = "port_ids_page_1.html";
+const PAGE2 = "port_ids_page_2.html";
+
+// Return the MIDI port id of the first input port for the given URL and page
+function id_for_tab(url, page) {
+  return BrowserTestUtils.withNewTab(
+    {
+      gBrowser,
+      url: url + page,
+      waitForLoad: true,
+    },
+    async function(browser) {
+      return SpecialPowers.spawn(browser, [""], function() {
+        return content.wrappedJSObject.get_first_input_id();
+      });
+    }
+  );
+}
+
+add_task(async function() {
+  let com_page1;
+  let com_page1_reload;
+  let org_page1;
+  let org_page2;
+
+  [com_page1, com_page1_reload, org_page1, org_page2] = await Promise.all([
+    id_for_tab(EXAMPLE_COM_URL, PAGE1),
+    id_for_tab(EXAMPLE_COM_URL, PAGE1),
+    id_for_tab(EXAMPLE_ORG_URL, PAGE1),
+    id_for_tab(EXAMPLE_ORG_URL, PAGE2),
+  ]);
+  Assert.equal(
+    com_page1,
+    com_page1_reload,
+    "MIDI port ids should be the same when reloading the same page"
+  );
+  Assert.notEqual(
+    com_page1,
+    org_page1,
+    "MIDI port ids should be different in different origins"
+  );
+  Assert.equal(
+    org_page1,
+    org_page2,
+    "MIDI port ids should be the same in the same origin"
+  );
+});